azkaban-common/src/test/java/azkaban/project/DirectoryFlowLoaderTest.java

```diff
@@ -65,4 +78,47 @@ public void testRecursiveLoadEmbeddedFlow() {
     // Should be 3 errors: jobe->innerFlow, innerFlow->jobe, innerFlow
     Assert.assertEquals(3, loader.getErrors().size());
   }
+
+  private static File decompressTarBZ2(InputStream is) throws IOException {
+    File outputDir = Files.createTempDir();
+
+    try (TarArchiveInputStream tais = new TarArchiveInputStream(new BZip2CompressorInputStream(is))){
+      TarArchiveEntry entry;
+      while ((entry = tais.getNextTarEntry()) != null) {
+        if (entry.isDirectory()) {
+          continue;
+        }
+
+        File outputFile = new File(outputDir, entry.getName());
+        File parent = outputFile.getParentFile();
+        if (!parent.exists()) {
+          parent.mkdirs();
+        }
+
+        IOUtils.copy(tais, new FileOutputStream(outputFile));
+      }
+
+      return outputDir;
+    }
+  }
+
+  @Test
+  public void testMassiveFlow() throws Exception {
+    final DirectoryFlowLoader loader = new DirectoryFlowLoader(new Props());
+
+    File projectDir = null;
+    try (InputStream is = Resources.getResource("project/testMassiveFlow/massiveFlows.tar.bz2").openStream()) {
```

Comment:
It's better to move the test data to /test/execution-test-data/. See PR #1361 for details.
 (https://github.com/azkaban/azkaban/pull/1361)